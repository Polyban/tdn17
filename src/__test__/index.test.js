import context from 'aws-lambda-mock-context'
import merge from 'deepmerge'
import { handler as lambda } from '../'
import launchRequest from '../__mock__/launch-request.json'
import informationRequest from '../__mock__/information-request.json'
import helpRequest from '../__mock__/help-request.json'
import cancelRequest from '../__mock__/cancel-request.json'

describe('Alexa TDN', () => {
  it('should handle LaunchRequest', () => {
    const ctx = context()
    lambda(launchRequest, ctx)
    return ctx.Promise.then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should invoke IntentRequest', () => {
    const ctx = context()
    lambda(informationRequest, ctx)
    return ctx.Promise.then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should invoke IntentRequest on new session', () => {
    const ctx = context()
    lambda(
      merge(
        informationRequest,
        { session: { new: true } }
      ),
      ctx
    )
    return ctx.Promise.then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should flow into Information and back', () => {
    const stepOne = () => {
      const ctx = context()
      lambda(launchRequest, ctx)
      return ctx.Promise
    }
    const stepTwo = () => {
      const ctx = context()
      lambda(informationRequest, ctx)
      return ctx.Promise
    }
    stepOne().then(stepTwo).then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should invoke help intent', () => {
    const ctx = context()
    lambda(helpRequest, ctx)
    return ctx.Promise.then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should exit when canceled from intro', () => {
    const stepOne = () => {
      const ctx = context()
      lambda(launchRequest, ctx)
      return ctx.Promise
    }
    const stepTwo = () => {
      const ctx = context()
      lambda(
        merge(
          cancelRequest,
          { session: { attributes: { STATE: 'INTRO' } } }
        ),
        ctx
      )
      return ctx.Promise
    }
    stepOne().then(stepTwo).then((response) => {
      expect(response).toMatchSnapshot()
    })
  })

  it('should not exit when canceled from intro', () => {
    const stepOne = () => {
      const ctx = context()
      lambda(informationRequest, ctx)
      return ctx.Promise
    }
    const stepTwo = () => {
      const ctx = context()
      lambda(
        merge(
          cancelRequest,
          { session: { attributes: { STATE: 'INFORMATION' } } }
        ),
        ctx
      )
      return ctx.Promise
    }
    stepOne().then(stepTwo).then((response) => {
      expect(response).toMatchSnapshot()
    })
  })
})
