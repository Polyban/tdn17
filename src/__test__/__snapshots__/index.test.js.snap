// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alexa TDN should exit when canceled from intro 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "text": "Auf Wiedersehen.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should flow into Information and back 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak><p>Der Tag der Niedersachsen 2017 findet zum 35. Mal statt. Nach 1983 und 2005 ist Wolfsburg nun zum dritten Mal Ausrichterstadt.</p><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should handle LaunchRequest 1`] = `
Object {
  "response": Object {
    "card": Object {
      "image": Object {
        "largeImageUrl": "https://tdn17.s3-eu-west-1.amazonaws.com/card.2x.png",
        "smallImageUrl": "https://tdn17.s3-eu-west-1.amazonaws.com/card.1x.png",
      },
      "text": "Sage \\"Informationen\\" für Allgemeine Informationen oder \\"Programm\\" für eine Programmübersicht.",
      "title": "Willkommen zum 35. Tag der Niedersachsen im Jahr 2017",
      "type": "Standard",
    },
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak><p>Willkommen zum 35. \\"Tag der Niedersachsen\\" im Jahr 2017. <break /> Ich habe allgemeine Informationen oder eine Programmübersicht.</p><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should invoke IntentRequest 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak><p>Der Tag der Niedersachsen 2017 findet zum 35. Mal statt. Nach 1983 und 2005 ist Wolfsburg nun zum dritten Mal Ausrichterstadt.</p><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should invoke IntentRequest on new session 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak><p>Der Tag der Niedersachsen 2017 findet zum 35. Mal statt. Nach 1983 und 2005 ist Wolfsburg nun zum dritten Mal Ausrichterstadt.</p><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should invoke help intent 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "
  Sage \\"Informationen\\" für allgemeine Informationen oder \\"Programm\\" für das Programm.

  Du kannst zum Beispiel das Programm für einen bestimmten Tag aufrufen, indem Du sagst:


  \\"Alexa, frage Tag der Niedersachsen nach dem Programm für Freitag\\"
",
      "title": "Hilfe",
      "type": "Simple",
    },
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak>Rufe Informationen oder das Programm ab, indem Du sagst: <break/> <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
      "type": "SSML",
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;

exports[`Alexa TDN should not exit when canceled from intro 1`] = `
Object {
  "response": Object {
    "directives": Array [],
    "outputSpeech": Object {
      "ssml": "<speak><p>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</p></speak>",
      "type": "SSML",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Sage <emphasis>Informationen</emphasis> oder <emphasis>Programm</emphasis>.</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": false,
  },
  "sessionAttributes": Object {
    "STATE": "INTRO",
  },
  "version": "1.0",
}
`;
